daemon/rename.go

```diff
@@ -21,6 +21,13 @@ func (daemon *Daemon) ContainerRename(oldName, newName string) error {
 		return fmt.Errorf("Neither old nor new names may be empty")
 	}
 
+	if newName[0] != '/' {
+		newName = "/" + newName
+	}
```

Comment:
I moved container.Lock up as there could be a chance where two requests "docker rename old new" and "docker rename old new" can land in parallel. We should fail the second request as after the first request there would not be a container by name old.